src/custom/state/orders/middleware.ts

```diff
@@ -224,11 +231,55 @@ export const soundMiddleware: Middleware<Record<string, unknown>, AppState> = (s
 }
 
 export const appziMiddleware: Middleware<Record<string, unknown>, AppState> = (store) => (next) => (action) => {
-  if (isBatchFulfillOrderAction(action) || isSingleFulfillOrderAction(action) || isExpireOrdersAction(action)) {
+  if (isBatchFulfillOrderAction(action)) {
     // Shows NPS feedback (or attempts to) when there's a successful trade
-    // Or the order has expired
-    openNpsAppziSometimes()
+    const {
+      chainId,
+      ordersData: [{ id }],
+    } = action.payload
+
+    const orders = store.getState().orders[chainId]
+    const openSince = _getOrderById(orders, id)?.order?.openSince
+
+    openNpsAppziSometimes({ traded: true, secondsSinceOpen: timeSinceInSeconds(openSince) })
+  } else if (isSingleFulfillOrderAction(action)) {
+    // Shows NPS feedback (or attempts to) when there's a successful trade
+    const { chainId, id } = action.payload
+
+    const orders = store.getState().orders[chainId]
+    const openSince = _getOrderById(orders, id)?.order?.openSince
+
+    openNpsAppziSometimes({ traded: true, secondsSinceOpen: timeSinceInSeconds(openSince) })
+  } else if (isBatchExpireOrderAction(action)) {
+    // Shows NPS feedback (or attempts to) when the order expired
+    const {
+      chainId,
+      ids: [id],
+    } = action.payload
+
+    const orders = store.getState().orders[chainId]
+    const openSince = _getOrderById(orders, id)?.order?.openSince
+
+    openNpsAppziSometimes({ expired: true, secondsSinceOpen: timeSinceInSeconds(openSince) })
+  } else if (isSingleExpireOrderAction(action)) {
+    // Shows NPS feedback (or attempts to) when the order expired
+    const { chainId, id } = action.payload
+
+    const orders = store.getState().orders[chainId]
+    const openSince = _getOrderById(orders, id)?.order?.openSince
+
+    openNpsAppziSometimes({ expired: true, secondsSinceOpen: timeSinceInSeconds(openSince) })
```

Comment:
Theres a bit of repetition in all this branches. 

Would it make sense to define before the `if` statement, a `let openNpsParams = Parameters<openNpsAppziSometimes> | undefined`

so if the param is set, you cal the NPS function

Reply:
https://github.com/cowprotocol/cowswap/pull/905/commits/354c7e89aebee7ffa2f36ef6266276dde76d3ddb